services/todo-legacy/application.ts

```diff
@@ -19,31 +18,24 @@ export class TodoApplication extends Application {
     app.controller(TodoController);
 
     let datasource = new DataSourceConstructor('ds', ds);
-    // FIXME(kev): datasource.createModel needs to return a
-    // juggler.PersistedModel instead of a juggler.ModelBase
-    // in order to use the DefaultCrudRepository.
-    let todo = require('./models/todo-model.json');
-    let TodoLegacy: any = datasource.createModel(todo.name, todo.properties);
-    let TodoRepository = new DefaultCrudRepository<Todo, number>(
-      TodoLegacy,
-      datasource
-    );
-    app.bind('repositories.todo').to(TodoRepository);
+    app.bind('datasources.ds').to(datasource);
```

Comment:
nitpick: if `ds` is a datasource that only `Todo` uses, IMO this logic can also be moved to file `todo-controller.ts`, then call repository decorator as:

```
let dsConfig = datasources['ds'];
let datasource = new DataSourceConstructor('ds', dsConfig);
... ...

export class TodoController {
  @repository(Todo, datasource)
  repository: EntityCrudRepository<Todo, number>;
  ... ...
}
```
I think it would be equal to your approach here.

Reply:
That defeats the purpose of dependency injection. Now my controller is tightly coupled to the repository implementation. If in the future, I want to inject a different instance, I'd have to change the controller code